frontend/src/features/admin-form/create/featureTour/FeatureTour.tsx

```diff
@@ -1,19 +1,36 @@
-import Joyride from 'react-joyride'
+import { useState } from 'react'
+import Joyride, { CallBackProps, EVENTS, STATUS } from 'react-joyride'
 import { useToken } from '@chakra-ui/react'
 
 import { FEATURE_STEPS } from './constants'
 import { FeatureTourTooltip } from './FeatureTourTooltip'
 
 interface FeatureTourProps {
-  shouldRun: boolean
+  onClose: () => void
 }
 
-export const FeatureTour = ({ shouldRun }: FeatureTourProps): JSX.Element => {
+export const FeatureTour = ({ onClose }: FeatureTourProps): JSX.Element => {
+  const [stepIndex, setStepIndex] = useState<number>(0)
   const arrowColor: string = useToken('colors', ['primary.100'])
+
+  const handleJoyrideCallback = (data: CallBackProps) => {
+    const { index, status, type } = data
+
+    if (type === EVENTS.STEP_AFTER || type === EVENTS.TARGET_NOT_FOUND) {
+      setStepIndex(index + 1)
+    }
+
+    if (status === STATUS.FINISHED || status === STATUS.SKIPPED) {
+      onClose()
+    }
```

Comment:
I don't think using a switch block is better because the conditions are checking for two different variables. Hence if we use a switch block, we would need to have 2 switch blocks (1 for `type` and 1 for `status`) and hence I am not very sure if that would be cleaner than this?

Reply:
oops i skimmed and didn't realize they were two variables. ignore this comment then!